packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -189,33 +178,35 @@ class Dart2JSTarget extends Dart2WebTarget {
       '--cfe-only',
       environment.buildDir.childFile('main.dart').path, // dartfile
     ];
-    globals.printTrace('compiling dart code to kernel with command "${compilationArgs.join(' ')}"');
+
+    final ProcessUtils processUtils = ProcessUtils(
+      logger: environment.logger,
+      processManager: environment.processManager,
+    );
 
     // Run the dart2js compilation in two stages, so that icon tree shaking can
     // parse the kernel file for web builds.
-    final ProcessResult kernelResult = await globals.processManager.run(compilationArgs);
-    if (kernelResult.exitCode != 0) {
-      throw Exception(_collectOutput(kernelResult));
-    }
+    await processUtils.run(compilationArgs, throwOnError: true);
 
     final File outputJSFile = environment.buildDir.childFile('main.dart.js');
 
-    final ProcessResult javaScriptResult = await environment.processManager.run(<String>[
-      ...sharedCommandOptions,
-      if (buildMode == BuildMode.profile) '--no-minify',
-      ...compilerConfig.toCommandOptions(),
-      '-o',
-      outputJSFile.path,
-      environment.buildDir.childFile('app.dill').path, // dartfile
-    ]);
-    if (javaScriptResult.exitCode != 0) {
-      throw Exception(_collectOutput(javaScriptResult));
-    }
-    final File dart2jsDeps = environment.buildDir
-      .childFile('app.dill.deps');
+    await processUtils.run(
+      throwOnError: true,
+      <String>[
+        ...sharedCommandOptions,
+        if (buildMode == BuildMode.profile) '--no-minify',
+        ...compilerConfig.toCommandOptions(),
+        '-o',
+        outputJSFile.path,
+        environment.buildDir.childFile('app.dill').path, // dartfile
+      ],
+    );
+    final File dart2jsDeps = environment.buildDir.childFile('app.dill.deps');
     if (!dart2jsDeps.existsSync()) {
-      environment.logger.printWarning('Warning: dart2js did not produced expected deps list at '
-        '${dart2jsDeps.path}');
+      environment.logger.printWarning(
+        'Warning: dart2js did not produced expected deps list at '
```

Comment:
nit: produced -> produce

Reply:
Ha! Uh. Exercise for later. This is just code reflow and not any change I made

Reply:
we should fix it now, it's trivial to fix and makes the product better :-)

Reply:
looks like this wasn't fixed in the version that landed?

Reply:
https://github.com/flutter/flutter/pull/126114